staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/validation/validation.go

```diff
@@ -55,6 +55,11 @@ func ValidateLabelSelectorRequirement(sr metav1.LabelSelectorRequirement, fldPat
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("operator"), sr.Operator, "not a valid selector operator"))
 	}
 	allErrs = append(allErrs, ValidateLabelName(sr.Key, fldPath.Child("key"))...)
+	for valueIndex, value := range sr.Values {
```

Comment:
this new bit is what needs to be wrapped in a `if validateLabelValues { ...` block